feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -271,37 +272,40 @@ private fun getStabilizationStringRes(
  * ON - Both preview and video are stabilized.
  * HIGH_QUALITY - Video will be stabilized, preview might be stabilized, depending on the device.
  * OFF - Preview and video stabilization is disabled.
+ *
+ * @param supportedStabilizationMode the enabled condition for this setting.
  */
 @Composable
 fun VideoStabilizeSetting(
     currentPreviewStabilization: Stabilization,
     currentVideoStabilization: Stabilization,
+    supportedStabilizationMode: SupportedStabilizationMode,
     setVideoStabilization: (Stabilization) -> Unit,
     setPreviewStabilization: (Stabilization) -> Unit
 ) {
     BasicPopupSetting(
         title = stringResource(R.string.video_stabilization_title),
         leadingIcon = null,
-        description = stringResource(
-            id = getStabilizationStringRes(
-                previewStabilization = currentPreviewStabilization,
-                videoStabilization = currentVideoStabilization
+        enabled = supportedStabilizationMode != SupportedStabilizationMode.UNSUPPORTED,
+        description = if (supportedStabilizationMode == SupportedStabilizationMode.UNSUPPORTED) {
+            stringResource(id = R.string.stabilization_description_unsupported)
+        } else {
+            stringResource(
+                id = getStabilizationStringRes(
+                    previewStabilization = currentPreviewStabilization,
+                    videoStabilization = currentVideoStabilization
+                )
             )
-        ),
+        },
         popupContents = {
             Column(Modifier.selectableGroup()) {
-                Text(
-                    text = stringResource(id = R.string.stabilization_popup_disclaimer),
-                    fontStyle = FontStyle.Italic,
-                    color = MaterialTheme.colorScheme.onPrimaryContainer
-                )
-
                 Spacer(modifier = Modifier.height(10.dp))
 
                 // on selector
                 SingleChoiceSelector(
                     text = stringResource(id = R.string.stabilization_selector_on),
                     secondaryText = stringResource(id = R.string.stabilization_selector_on_info),
+                    enabled = supportedStabilizationMode == SupportedStabilizationMode.FULL,
                     selected = (currentPreviewStabilization == Stabilization.ON) &&
                         (currentVideoStabilization != Stabilization.OFF),
                     onClick = {
```

Comment:
Technically setting both VideoStabilization and PreviewStabilization to `ON` will work here, but only because CameraX will internally ignore the VideoStabilization part.

Let's set VideoStabilization to `UNDEFINED` to better represent what's actually going on under the hood and so we don't confuse developers into thinking they have to set both to `ON` to get preview stabilization.